### PR TITLE
Integrate the OpmphmPredictor

### PR DESCRIPTION
**Pull request by KurtMi**
also adds new ksLookup flag `KDB_O_BINSEARCH` and a new 
benchmarks of the OPMPHM Prediction Time

## Basics

Do not describe the purpose here but:

- [x] Short descriptions should be in the release notes (added as entry in
      doc/news/_preparation_next_release.md which contains `*(my name)*`)
      **Please always add something to the the release notes.**
- [x] Longer descriptions should be in documentation or in design decisions.
- [x] Describe details of how you changed the code in commit messages
      (first line should have `module: short statement` syntax)
- [ ] References to issues, e.g. `close #X`, should be in the commit messages.

## Checklist

Check relevant points but **please do not remove entries**.
For docu fixes, spell checking, and similar none of these points below
need to be checked.

- [x] I added unit tests
- [x] I ran all tests locally and everything went fine
- [x] affected documentation is fixed
- [x] I added code comments, logging, and assertions (see doc/CODING.md)
- [ ] meta data is updated (e.g. README.md of plugins)

## Review

### CODE REVIEW

**Comment by KurtMi** [comment]
jenkins build all please

**Comment by KurtMi** [comment]
jenkins build all please

**Comment by KurtMi** [comment]
jenkins build all please

**Comment by KurtMi** [comment]
jenkins build all please

**Comment by KurtMi** [comment]
jenkins build all please

**Comment by KurtMi** [comment]
@markus2330 please review my last PR

File: doc/news/_preparation_next_release.md [hunk 1]
```diff
@@ -61,13 +61,21 @@ You can also read the news [on our website](https://www.libelektra.org/news/0.8.
...
 - <<HIGHLIGHT2>>
 - <<HIGHLIGHT3>>
 
 
-### <<HIGHLIGHT1>>
+### Hybrid Search Algorithm for `ksLookup (...)`
 
+The hybrid search algorithm is now implemented, this concludes the extension of the `ksLookup (...)` search with the
+[order preserving minimal perfect hash map (OPMPHM)](https://master.libelektra.org/doc/dev/data-structures.md#order-preserving-minimal-perfect-hash-map-aka-opmphm).
+The hybrid search combines the best properties of the binary search and the [OPMPHM](https://master.libelektra.org/doc/dev/data-structures.md#order-preserving-minimal-perfect-hash-map-aka-opmphm).
```
Comment (markus2330) [inline doc/news/_preparation_next_release.md]: Can you add some numbers here?

File: doc/news/_preparation_next_release.md [hunk 2]
```diff
@@ -61,13 +61,21 @@ You can also read the news [on our website](https://www.libelektra.org/news/0.8.
...
+The hybrid search decides dynamically which search algorithm to use, the API user can overrule the hybrid search by passing
```
Comment (markus2330) [inline doc/news/_preparation_next_release.md]: You should note that the constants are only in kdbproposal.h

**Pull request merged**